### PR TITLE
NETOBSERV-1498 Add shellcheck linter for bash and make help target

### PR DESCRIPTION
**Pull request by msherif1234**
Add shellcheck bash scripts linter and fix linter error
Add make help target

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,24 +21,30 @@ prereqs: ## Test if prerequisites are met, and installing missing dependencies
...
-image:
+image: ## Build the docker image
 	$(OCI_BIN) build -t network-observability-cli .
 
 .PHONY: lint
 lint: prereqs ## Lint code
 	@echo "### Linting code"
 	golangci-lint run ./...
+	@echo "### Run shellcheck against bash scripts"
+	find . -name '*.sh' | xargs shellcheck
```
Comment (jpinsonneau) [inline Makefile:38]: Is there a way to install `shellcheck` automatically in the `prereqs` target ?

File: Makefile [hunk 2]
```diff
@@ -21,24 +21,30 @@ prereqs: ## Test if prerequisites are met, and installing missing dependencies
...
+clean: ## Clean up build directory
 	@rm -rf $(DIST_DIR)
 
 .PHONY: oc-commands
 oc-commands: build
 	sudo cp -a ./build/. /usr/bin/
+
+.PHONY: help
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
```
Comment (jpinsonneau) [inline Makefile:50]: :+1:

File: res/functions.sh [hunk 3]
```diff
@@ -1,3 +1,6 @@
+#!/usr/bin/env bash
+set -eux
+
```
Comment (jpinsonneau) [inline res/functions.sh:3]: :+1: 

Don't spend too much effort on the scripts, I have pending PRs that changes everything in these:
https://github.com/netobserv/network-observability-cli/pull/3
https://github.com/netobserv/network-observability-cli/pull/4

We should consider merging them before fixing

File: Makefile (hunk 1 above)
Comment (jpinsonneau) [reply Makefile:38]: https://developer.ibm.com/blogs/shellcheck-and-you-should-too/

Comment (msherif1234) [reply Makefile:38]: no this a pkg u need to install on ur host and it depends on the os

File: res/functions.sh (hunk 3 above)
Comment (msherif1234) [reply res/functions.sh:3]: ok since this repo has bash dependency I thought having lint for bash will be add good value

File: Makefile (hunk 1 above)
Comment (jpinsonneau) [reply Makefile:38]: the article above suggest:
```
  docker pull koalaman/shellcheck:latest
  docker run -v "$PWD:/mnt" koalaman/shellcheck "$1"
```

The advantage of this is that `shellcheck` will not be a requirement for devs / CI

Else we need to add that in the README

Comment (msherif1234) [reply Makefile:38]: updated readme its not worth spinning up docker atm for single pkg dependency

**Review by jpinsonneau — APPROVED** [review]
LGTM, feel free to merge @msherif1234 :smile_cat:

**Pull request merged**